src/lab.rs

```diff
@@ -190,33 +198,75 @@ pub fn test_mutants(
     Ok(lab_outcome)
 }
 
-fn test_timeout(baseline_test_duration: Option<Duration>, options: &Options) -> Duration {
-    if let Some(timeout) = options.test_timeout {
+#[derive(Copy, Clone)]
+struct Timeouts {
+    build: Duration,
+    test: Duration,
+}
+
+fn phase_timeout(
+    explicit_timeout: Option<Duration>,
+    check_only_timeout: Duration,
+    baseline_duration: Option<Duration>,
+    minimum: Duration,
+    multiplier: f64,
+    phase_name: &str,
+    options: &Options,
+) -> Duration {
+    if let Some(timeout) = explicit_timeout {
         timeout
     } else if options.check_only {
-        Duration::ZERO
-    } else if let Some(baseline_test_duration) = baseline_test_duration {
-        let timeout = max(
-            options.minimum_test_timeout,
-            Duration::from_secs(
-                (baseline_test_duration.as_secs_f64()
-                    * options.test_timeout_multiplier.unwrap_or(5.0))
-                .round() as u64,
-            ),
+        check_only_timeout
+    } else if let Some(baseline_duration) = baseline_duration {
```

Comment:
I think somewhere in here we want to say, "if Options.in_place && phase == Build|Check" then don't use the baseline, give a warning and use the minimum.

That might be best done by giving this a phase enum parameter and folding the `build/test_timeout` functions into this, but see what works.

Reply:
Good point, I didn't know that the baseline can be computed in-place.